moabb/datasets/neiry.py

```diff
@@ -0,0 +1,188 @@
+            ("stimuli", np.object),
+        ]
+    )
+
+    def __init__(self):
+        super().__init__(
+            subjects=list(range(60)),
+            sessions_per_subject=17,
+            events={"Target": 1, "NonTarget": 2},
+            code="Demons P300",
+            interval=[0, 1],
+            paradigm="p300",
+        )
+        self.path = None
+        self.subjects_filenames = None
+
+    @staticmethod
+    def _strip(session) -> tuple:
+        """Strips nans (from right side of all channels) added during hdf5 packaging
+
+        Returns:
+            tuple ready to be converted to `_session_dtype`
+        """
+        eeg, *rest = session
+        ind = -next(i for i, value in enumerate(eeg[0, ::-1]) if not np.isnan(value))
+        if ind == 0:
+            ind = None
+        return tuple((eeg[:, :ind], *rest))
+
+    @classmethod
+    def read_hdf(cls, filename) -> np.ndarray:
+        """Reads data from HDF file
+
+        Returns:
+            array of `_act_dtype`
+        """
+        with h5py.File(filename, "r") as hfile:
+            group = hfile[cls._hdf_path]
+            record = np.empty(len(group), cls._act_dtype)
+            for i, act in enumerate(group.values()):
+                record[i]["sessions"] = np.array(
+                    [cls._strip(item) for item in act], cls._session_dtype
+                )
+                for name, value in act.attrs.items():
+                    record[i][name] = value
+        return record
+
+    def _get_single_subject_data(self, subject: int):
+        record = self.read_hdf(self.data_path(subject))
+
+        info = create_info(
+            self.ch_names + ["mult_target", "target"],
+            self.sampling_rate,
+            ["eeg"] * len(self.ch_names) + ["misc", "stim"],
+        )
+        montage = make_standard_montage("standard_1020")
+
+        runs_raw = {}
+        for i, act in enumerate(record):
+            # target and stims are increased by 1
+            # because the channel is filled with zeros by default
+            target = act["target"] + 1
+            run_data = []
+            for eeg, starts, stims in act["sessions"]:
+                starts = starts * self.sampling_rate / self._ms_in_sec
+                starts = starts.round().astype(np.int)
+                stims = stims + 1
+                stims_channel = np.zeros(eeg.shape[1])
+                target_channel = np.zeros(eeg.shape[1])
+
+                for start, stimul in zip(starts, stims):
+                    stims_channel[start] = stimul
+                    target_channel[start] = 1 if stimul == target else 2
+
+                round_data = np.vstack(
+                    (eeg, stims_channel[None, :], target_channel[None, :])
+                )
+                run_data.append(round_data)
+
+            raw = RawArray(np.hstack(run_data), info)
+            raw.set_montage(montage)
+            runs_raw[f"run_{i}"] = raw
+        return {"session_0": runs_raw}
```

Comment:
You should update this, as you have 17 sessions.

Reply:
No, this is one session. I mean person put helmet only once. All runs happens in ~15 minutes.
So one session is what suitable here as far as I understand

Reply:
I first thought I put records to different sessions but that's not true =)

Reply:
No problem, in that case you need to update line 96 to indicate that there is one session.

Reply:
oh, I see